dmutils/content_loader.py

```diff
@@ -314,68 +315,174 @@ def unformat_data(self, data):
+        return bool(question) and question.get('assuranceApproach', False)
 
-    def _is_list_type(self, key):
-        """Return True if a given key is a list type"""
-        return key == 'serviceTypes' or self._is_type(key, 'list', 'checkboxes')
 
-    def _is_not_upload(self, key):
-        """Return True if a given key is not a file upload"""
-        return not self._is_type(key, 'upload')
+class ContentQuestion(object):
+    def __init__(self, data, number=None):
+        self.number = number
+        self._data = data.copy()
 
-    def _is_boolean_type(self, key):
-        """Return True if a given key is a boolean type"""
-        return self._is_type(key, 'boolean')
+        if 'questions' in data:
+            self.questions = [
+                question if isinstance(question, ContentQuestion) else ContentQuestion(question)
+                for question in data['questions']
+            ]
+        else:
+            self.questions = None
 
-    def _is_numeric_type(self, key):
-        """Return True if a given key is a numeric type"""
-        return self._is_type(key, 'percentage')
+    def summary(self, service_data):
+        return ContentQuestionSummary(
+            self, service_data
+        )
 
-    def _is_pricing_type(self, key):
-        """Return True if a given key is a pricing type"""
-        return self._is_type(key, 'pricing')
+    def get_question(self, field_name):
+        if self.id == field_name:
+            return self
+        elif self.questions:
+            return next(
+                (question for question in self.questions if question.id == field_name),
+                None
+            )
 
-    def _has_assurance(self, key):
-        """Return True if a question has an assurance component"""
-        question = self.get_question(key)
-        return bool(question) and question.get('assuranceApproach', False)
+    def get_data(self, form_data):
+        if self.id in form_data and self.type == 'pricing':
+            return expand_pricing_field(form_data.getlist(self.id))
+        elif self.questions:
+            questions_data = {}
+            for question in self.questions:
+                questions_data.update(question.get_data(form_data))
+            return questions_data
+        else:
+            return self._get_single_question_data(form_data)
 
+    def _get_single_question_data(self, form_data):
+        if self.id not in form_data:
+            if self.get('assuranceApproach') and '{}--assurance'.format(self.id) in form_data:
+                return {self.id: {'assurance': form_data.get('{}--assurance'.format(self.id))}}
+            else:
+                return {}
+
+        if self.id == 'serviceTypes' or self.type in ['list', 'checkboxes']:
+            value = form_data.getlist(self.id)
+        elif self.type in ['boolean']:
+            value = convert_to_boolean(form_data[self.id])
+        elif self.type in ['percentage']:
```

Comment:
It could be. I kept `in` checks because they're easier to add to (if we might get more boolean/numeric types) and keep the same pattern with `['list', 'checkboxes']`, but I don't think it matters too much